### PR TITLE
fix(@desktop/wallet): Improving performance when switching accounts

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

### Cool Spaceship Picture

<!-- optional but cool ->

### CODE REVIEW

File: ui/imports/shared/views/AssetsView.qml [hunk 1]
```diff
@@ -108,16 +108,28 @@ ColumnLayout {
...
                         }
                         else {
-                            return totalBalance
+                            return model.currentBalance
                         }
                     }
-                    expectedRoles: ["balances", "marketDetails", "decimals", "communityId"]
+                    expectedRoles: ["marketDetails", "communityId", "currentBalance"]
                 },
+//                FastExpressionRole {
```
Comment (caybro) [inline ui/imports/shared/views/AssetsView.qml:120]: Why the commented out block? :)

File: ui/imports/shared/views/AssetsView.qml [hunk 2]
```diff
@@ -108,16 +108,28 @@ ColumnLayout {
                 FastExpressionRole {
                     name: "currentCurrencyBalance"
                     expression: {
-                        let totalBalance = d.getTotalBalance(model.balances, model.decimals, root.addressFilters, root.networkFilters)
```
Comment (caybro) [inline ui/imports/shared/views/AssetsView.qml:111]: This turns out to be the culprit? Afaik I warned exactly against this :)

Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/AssetsView.qml:111]: actually no its doesnt solve the problem, I guess i'll test with Michals fast sorters and filters and see if it improves

**Review by alaibe — APPROVED** [review]

**Review by stefandunca — APPROVED** [review]
🤩

File: ui/imports/shared/views/AssetsView.qml [hunk 3]
```diff
@@ -77,12 +77,12 @@ ColumnLayout {
             submodelRoleName: "balances"
             delegateModel: SortFilterProxyModel {
                 sourceModel: submodel
-                filters: ExpressionFilter {
+                filters: FastExpressionFilter {
                     expression: {
                         root.networkFilters
```
Comment (caybro) [inline ui/imports/shared/views/AssetsView.qml:82]: Missing the `expectedRoles` property

**Review by caybro — CHANGES_REQUESTED** [review]
Somehow I think we should make the `expectedRoles` property `REQUIRED` on the C++ level, wdyt @micieslak ?

**Pull request merged**